File: code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/tile/CustomTileFilter.java
package org.eclipse.scout.widgets.client.ui.tile;

import org.eclipse.scout.rt.client.ui.tile.ITileFilter;
import org.eclipse.scout.rt.platform.util.ObjectUtility;

public class CustomTileFilter implements ITileFilter<ICustomTile> {
  private String m_text;

  public CustomTileFilter() {
    setText("");
  }

  public void setText(String text) {
    text = ObjectUtility.nvl(text, "");
    m_text = text.toLowerCase();
  }

  @Override
  public boolean accept(ICustomTile tile) {
    String label = ObjectUtility.nvl(tile.getLabel(), "");
    String filterText = label.trim().toLowerCase();
    return filterText.indexOf(m_text) > -1;
  }

}

File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/jswidgets/tilegrid/CustomTileFilter.js
jswidgets.CustomTileFilter = function(model) {
  model = model || {};
  this.text = null;
  this.setText(model.text);
};

jswidgets.CustomTileFilter.prototype.setText = function(text) {
  this.text = text || '';
  this.text = this.text.toLowerCase();
};

jswidgets.CustomTileFilter.prototype.accept = function(tile) {
  var label = '';
  if (tile instanceof jswidgets.CustomTile) {
    label = tile.label || '';
  } else if (tile instanceof scout.HtmlTile) {
    label = scout.strings.plainText(tile.content) || '';
  }
  var filterText = label.trim().toLowerCase();
  return filterText.indexOf(this.text) > -1;
};
